NICE.Identity.Management/ClientApp/src/views/DeleteOrganisation/DeleteOrganisation.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from "react";
+import { RouteComponentProps, Link } from "react-router-dom";
+
+import { Alert } from "@nice-digital/nds-alert";
+import { Breadcrumbs, Breadcrumb } from "@nice-digital/nds-breadcrumbs";
+import { Button } from "@nice-digital/nds-button";
+import { PageHeader } from "@nice-digital/nds-page-header";
+
+import { DeleteOrganisationConfirmation } from "../DeleteOrganisationConfirmation/DeleteOrganisationConfirmation";
+import { OrganisationType } from "../../models/types";
+import { Endpoints } from "../../data/endpoints";
+import { ErrorMessage } from "../../components/ErrorMessage/ErrorMessage";
+
+import { fetchData } from "../../helpers/fetchData";
+import { isDataError } from "../../helpers/isDataError";
+
+type TParams = { id: string };
+
+type DeleteOrganisationProps = Record<string, unknown> & RouteComponentProps<TParams>;
+
+type DeleteOrganisationState = {
+	organisation: OrganisationType;
+	hasBeenDeleted: boolean;
+	error?: Error;
+	isLoading: boolean;
+	isDeleteButtonLoading: boolean;
+};
+
+export class DeleteOrganisation extends Component<DeleteOrganisationProps, DeleteOrganisationState> {
+	constructor(props: DeleteOrganisationProps) {
+		super(props);
+
+		this.state = {
+			organisation: {} as OrganisationType,
+			hasBeenDeleted: false,
+			isLoading: true,
+			isDeleteButtonLoading: false,
+		};
+	}
+
+	handleDeleteClick = async (): Promise<void> => {
+		this.setState({ isDeleteButtonLoading: true });
+
+		const fetchOptions = {
+			method: "DELETE",
+			headers: { "Content-Type": "application/json" },
+			body: JSON.stringify({ id: this.props.match.params.id }),
+		};
+
+		const deletedOrganisation = await fetchData(
+			Endpoints.organisation(this.props.match.params.id),
+			fetchOptions,
+		);
+
+		if (isDataError(deletedOrganisation)) {
+			this.setState({ error: deletedOrganisation });
+		} else {
+			this.setState({ hasBeenDeleted: true });
+		}
+
+		this.setState({ isDeleteButtonLoading: false });
+	};
+
+	async componentDidMount(): Promise<void> {
+		this.setState({ isLoading: true });
+
+		const organisation = await fetchData(Endpoints.organisation(this.props.match.params.id));
+
+		if (isDataError(organisation)) {
+			this.setState({ error: organisation });
+		}
+
+		this.setState({ organisation, isLoading: false });
+		document.title = `NICE Accounts - Delete ${organisation.name}`;
```

Comment:
This should go in the bottom of the constructor above.

Reply:
Leaving it in component did mount as constructor doesn't have org name yet